moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1728,6 +1799,17 @@ bool PlanningScene::processCollisionObjectMsg(const moveit_msgs::CollisionObject
     }
     if (!object.type.key.empty() || !object.type.db.empty())
       setObjectType(object.id, object.type);
+    // Add named frames
+    std::map<std::string, Eigen::Affine3d> named_frames;
+    Eigen::Affine3d p;
+    for (std::size_t i = 0; i < object.named_frames.size(); ++i)
+    {
+      tf::poseMsgToEigen(object.named_frames[i], p);
+      std::string name = object.frame_names[i];
+      named_frames[name] = t * p;
```

Comment:
This is interesting, can you explain why you chose to transform the named_frames into the world frame rather than leaving them wrt their parent frame? It seems like it may be less work to keep the transforms as "the tip is some pose from the object" rather than have to update the tip transform every time you move the object in the planning scene.

Reply:
It was so that getFrameTransform() and knowsFrameTransform() would apply the same way as they do to the other transforms in the world. I just copy-pasted it from the lines above, because I wanted to change as little as possible. I agree, it looks inefficient, but the same argument may apply to the other poses (primitives/meshes/planes).

But it is also true that the other poses are relevant for collision checking, and these frames are not, so keeping the named frames in the object reference frame might be better. I don't know if that makes a difference though - I haven't read the collision checking code.